File: 00_ETC/14995.cpp
#include <iostream>
using namespace std;

long long fibonacci(int n) {
    if (n <= 3) {
        return 1;
    }
    long long f1 = 1, f2 = 1, f3 = 1, result = 1;

    for (int i = 4; i <= n; ++i) {
        result = f3 + f1;
        f1 = f2;
        f2 = f3;
        f3 = result;
    }
    return result;
}

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);
    cout.tie(NULL);

    int n;
    cin >> n;
    cout << fibonacci(n) << endl;
    return 0;
}
//
// Created by 최승우 on 25. 8. 26.
//

File: 00_ETC_PY/14995.py
n = int(input())


def fibonacci(n):
    if (n <= 3):
        return 1
    f1 = 1
    f2 = 1
    f3 = 1
    result = 1

    for i in range(4, n + 1):
        result = f1 + f3
        f1 = f2
        f2 = f3
        f3 = result
    return result


print(fibonacci(n))